src/rulesets/oas3/ruleset.json

```diff
@@ -2388,18 +2388,31 @@
         }
       }
     },
-    "valid-example": {
-      "summary": "Examples must be valid against their defined schema.",
+    "valid-schema-example": {
+      "summary": "Schema examples must be valid against their defined schema.",
       "message": "\"{{property}}\" property {{error}}",
       "type": "validation",
-      "given": "$..[?(@.example)]",
+      "given": "$..*[?(@.schema)][?(@.example)]",
       "then": {
         "function": "schemaPath",
         "functionOptions": {
           "field": "example",
           "schemaPath": "$"
         }
       }
+    },
+    "valid-openapi-example": {
+      "summary": "Examples must be valid against ???",
+      "message": "{{error}}",
+      "type": "validation",
+      "given": "$..*[?(!@.schema)][?(@.example)]",
+      "then": {
+        "function": "example",
```

Comment:
I've created a separate function for this validation - `example`. But this is only for prototyping purposes.